File: 53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_sum, max_sum = 0, nums[0]
        for num in nums:
            curr_sum += num
            max_sum = max(curr_sum, max_sum)
            if(curr_sum < 0):
                curr_sum = 0
        return max_sum

File: 53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int l = 0, sum = 0, max_sum = nums[0];
        for(int i = 0; i < nums.size(); i++) {
            sum += nums[i];
            max_sum = max(max_sum, sum);
            while(sum < 0) {
                sum -= nums[l];
                l += 1;
            }
        }
        return max_sum;
    }
};